### PR TITLE
move syntax tests to unit tests

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
bors r+
🤖

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 cache: cargo
 before_cache:
-    # ask Cargo to clean up artifacts for workspace crates
-    - cargo clean $(printf -- '--package %s ' $(cd crates; ls))
-    # Cargo doesn't clean up depinfo itself, do it manually
-    - find ./target/debug -maxdepth 2 -iname '*.d' -delete
+    - find ./target/debug -maxdepth 1 -type f -delete
+    - rm -fr ./target/debug/{deps,.fingerprint}/{*ra_*,*heavy_test*,*gen_lsp*,*thread_worker*}
```
Comment (lnicola) [inline .travis.yml]: Did you mean to undo this change?

Comment (matklad) [reply .travis.yml]: Yeah, turns out `cargo` doesn't clean everything (.rlib files are still there for some of the workspace's crates), so the old way seems more robust.

Comment (lnicola) [reply .travis.yml]: But it still cleans `ra_vfs`, right?

Comment (matklad) [reply .travis.yml]: yes
